core/server/controllers/frontend/index.js

```diff
@@ -67,7 +67,10 @@ frontendControllers = {
             }
 
             // CASE: we only support /:slug format for pages
-            if (post.page && post.url !== req.path) {
+            // However, given that a url with `/edit/` appended is different
+            // than the post url, we must remove `/edit/` from the url if it
+            // is an edit url.
+            if (post.page && post.url !== ((lookup.isEditURL) ? req.path.replace(/edit\/$/, '') : req.path)) {
```

Comment:
That was my initial approach. Unfortunately though - that means that in theory if a user had permalinks turned on, both `blog.com/2016/08/05/page-name/edit` and `blog.com/page-name/edit` would work.

There was an explicit unit test in there already for that case that failed with just switching the order - so I went with this method :)